src/Hintman/Config.hs

```diff
@@ -44,9 +44,14 @@ hintmanConfiguationCodec = HintmanConfig
 
 loadFileConfig :: FilePath -> IO HintmanConfig
 loadFileConfig path =
-    Toml.decodeFile hintmanConfiguationCodec path `catch` catchLoad
+    (Toml.decodeFile hintmanConfiguationCodec path `catch` catchExcAndReturnDefaultConfig)
+      `catch` catchExcAndReturnDefaultConfig
   where
-    catchLoad :: LoadTomlException -> IO HintmanConfig
-    catchLoad _ = do
-        print $ "Hintman failed to load config from " <> path
-        pure defaultHintmanConfig
+    catchExcAndReturnDefaultConfig :: SomeException -> IO HintmanConfig
+    catchExcAndReturnDefaultConfig e = case e of
+        Exc (_ :: IOException) -> do
+            print $ "Config '" <> path <> "' does not exist! Using default configuration"
+            pure defaultHintmanConfig
+        _ -> do
+            print $ "Hintman failed to load config from " <> path
+            pure defaultHintmanConfig
```

Comment:
I see the pattern in these two cases. Not to repeat yourself you probably can move the text to print into the function in the where clause and the body the will look like:

```haskell
print errorMessage
pure defaultHintmanConfig
```

or just
```haskell
defaultHintmanConfig <$ print errorMessage
```

Reply:
:+1: